client/main.lua

```diff
@@ -336,53 +336,96 @@ RegisterNetEvent("qb-garages:client:VehicleList", function(data)
     end, indexgarage, type, garage.vehicle)
 end)
 
-RegisterNetEvent('qb-garages:client:takeOutGarage', function(data)
+local function SetProperties(veh, props, heading, owner)
+    QBCore.Functions.TriggerCallback('qb-garage:server:GetVehicleProperties', function(properties)
+        if props.plate then
+            SetNetworkIdAlwaysExistsForPlayer(NetworkGetNetworkIdFromEntity(veh), PlayerPedId(), true)
+            TriggerServerEvent('qb-garage:server:UpdateOutsideVehicle', props.plate, NetworkGetNetworkIdFromEntity(veh))
+        end
+
+        QBCore.Functions.SetVehicleProperties(veh, properties)
+        SetVehicleNumberPlateText(veh, props.plate)
+        SetEntityHeading(veh, heading)
+        exports['LegacyFuel']:SetFuel(veh, props.fuel)
+        doCarDamage(veh, props)
+        SetEntityAsMissionEntity(veh, true, true)
+        if not owner then
+            TaskWarpPedIntoVehicle(PlayerPedId(), veh, -1)
+        else
+            local playerIdx = GetPlayerFromServerId(owner)
+            local ped = GetPlayerPed(playerIdx)
+            TaskWarpPedIntoVehicle(ped, veh, -1)
+        end
+        SetVehicleEngineOn(veh, true, true)
+    end, props.plate)
+
+end
+
+RegisterNetEvent('qb-garage:client:SetProperties', function(netId, vehicle, heading, owner)
+    if not NetworkDoesEntityExistWithNetworkId(netId) then
+        print("Vehicle not found")
+        -- vehicles wont instantly exist on the client, even though they exist on the server.
+        Wait(500)
+    end
+    if NetworkDoesEntityExistWithNetworkId(netId) then
+        print("Found id : ".. netId .. "applying props")
+        local veh = NetworkGetEntityFromNetworkId(netId)
+        SetProperties(veh, vehicle, heading, owner)
+    else
+        print("Could not find id : ".. netId)
```

Comment:
I think it's better if you use `error` instead of `print` so this gets more attention